packages/react-aria-components/src/Link.tsx

```diff
@@ -0,0 +1,98 @@
+/*
+ * Copyright 2022 Adobe. All rights reserved.
+ * This file is licensed to you under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License. You may obtain a copy
+ * of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under
+ * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ * OF ANY KIND, either express or implied. See the License for the specific language
+ * governing permissions and limitations under the License.
+ */
+
+import {AriaLinkOptions, mergeProps, useFocusRing, useHover, useLink} from 'react-aria';
+import {ContextValue, RenderProps, SlotProps, useContextProps, useRenderProps} from './utils';
+import {mergeRefs} from '@react-aria/utils';
+import React, {createContext, ForwardedRef, forwardRef, useMemo} from 'react';
+
+export interface LinkProps extends Omit<AriaLinkOptions, 'elementType'>, RenderProps<LinkRenderProps>, SlotProps {}
+
+export interface LinkRenderProps {
+  /**
+   * Whether the link is the current item within a list.
+   * @selector [aria-current]
+   */
+  isCurrent: boolean,
+  /**
+   * Whether the link is currently hovered with a mouse.
+   * @selector [data-hovered]
+   */
+  isHovered: boolean,
+  /**
+   * Whether the link is currently in a pressed state.
+   * @selector [data-pressed]
+   */
+  isPressed: boolean,
+  /**
+   * Whether the link is focused, either via a mouse or keyboard.
+   * @selector :focus
+   */
+  isFocused: boolean,
+  /**
+   * Whether the link is keyboard focused.
+   * @selector [data-focus-visible]
+   */
+  isFocusVisible: boolean,
+  /**
+   * Whether the link is disabled.
+   * @selector [aria-disabled]
+   */
+  isDisabled: boolean
+}
+
+export const LinkContext = createContext<ContextValue<LinkProps, HTMLAnchorElement>>(null);
+
+function Link(props: LinkProps, ref: ForwardedRef<HTMLAnchorElement>) {
+  [props, ref] = useContextProps(props, ref, LinkContext);
+
+  let elementType = typeof props.children === 'string' ? 'span' : 'a';
```

Comment:
we handle elementType here, so should be able to handle it up in Button in the same way?

or would it make more sense for us to control the rendering of the tag instead of doing the whole clone element thing?

Reply:
I prefer to enforce an element type if we can. Router links are just too common though so had to support it here